src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -166,7 +166,8 @@ void AppHost::_HandleCommandlineArgs(const Remoting::WindowRequestedArgs& window
 
                 if (_windowLogic.ShouldExitEarly())
                 {
-                    ExitThread(result);
+                    TerminateProcess(GetCurrentProcess(), gsl::narrow_cast<UINT>(result));
+                    __assume(false);

```

Comment:
...this bit. I'm not entirely sure what this `ShouldExitEarly` branch is supposed to do here. Is it like the `wt.exe` flag and should just `PostQuitMessage` here? In any case, the `ExitThread` call was definitely wrong since it leaks any and all resources without cleanup.

Reply:
I don't know if this is quite right. I don't know whose job it is to pop up the MessageBox with the usage information for `wt --help` (for example), but if it's the one WindowsTerminal.exe process that is hosting all the windows ... then that will pose a problem.

Check whether `wtd --help` causes all existing Terminal Dev windows to terminate!